### PR TITLE
Update XYChart to handle partial dimensions

### PR DESCRIPTION
**Pull request by kyythane**
Our charts require that we make updates to the chart's margin after the initial layout. Currently, that can be a little awkward as `XYChart` resets the margin on each render. This change 1) allows disabling `XYChart` from updating the `margin` by passing in `null` and 2) updates `useDimensions` so it can handle partial objects, so you can update only 1 dimension or margin without needing to keep track of the rest.

#### :boom: Breaking Changes

-

#### :rocket: Enhancements

- DataProvider now takes a Partial Dimensions object.

#### :memo: Documentation

-

#### :bug: Bug Fix

-

#### :house: Internal

-

### CODE REVIEW

**Comment by kyythane** [comment]
If the merge commits pose an issue, I can reopen this PR off of a clean copy of the repo.

File: packages/visx-xychart/src/components/XYChart.tsx [hunk 1]
```diff
@@ -168,7 +167,23 @@ export default function XYChart<
...
+}
+
+function XYChartInternal(props: {
+  captureEvents: boolean;
+  accessibilityLabel: string;
+  children: React.ReactNode;
+}) {
+  const { captureEvents, accessibilityLabel, children } = props;
+  const eventEmitters = useEventEmitters({ source: XYCHART_EVENT_SOURCE });
+  const { width, height, margin } = useContext(DataContext);
```
Comment (kyythane) [inline packages/visx-xychart/src/components/XYChart.tsx:184]: Created an internal component so I could read `width`, `height`, and `margin` off of the `DataContext`. Doing that in one component felt a little convoluted? 

This also moved `useEventEmitters`. Is there an issue if `useEventEmitters` is called after `useEventHandlers`?

Comment (kyythane) [reply packages/visx-xychart/src/components/XYChart.tsx:184]: I can't check, but I _believe_ this extra component is what is causing happo to fail?

File: packages/visx-xychart/src/hooks/useDimensions.ts [hunk 2]
```diff
@@ -7,29 +7,45 @@ const DEFAULT_DIMS = {
...
-): [Dimensions, (dims: Dimensions) => void] {
+  initialDims: PartialDimensions = {},
+): [Dimensions, (dims: PartialDimensions) => void] {
   const [dimensions, privateSetDimensions] = useState<Dimensions>({
-    width: initialDims?.width == null ? DEFAULT_DIMS.width : initialDims.width,
-    height: initialDims?.height == null ? DEFAULT_DIMS.height : initialDims.height,
-    margin: initialDims?.margin == null ? DEFAULT_DIMS.margin : initialDims.margin,
+    // using || instead of ?? to include 0 as falsy
+    width: initialDims.width || DEFAULT_DIMS.width,
+    height: initialDims.height || DEFAULT_DIMS.height,
```
Comment (kyythane) [inline packages/visx-xychart/src/hooks/useDimensions.ts:20]: With this change, I wonder if `DEFAULT_DIMS` should be updated to match the default in XYChart? If so, I think I can remove the `null` from the margin type and maintain current behavior.

**Comment by kyythane** [comment]
@williaster quick ping, sorry if you're busy. Curious about your thoughts on this PR

File: packages/visx-xychart/src/components/XYChart.tsx [hunk 3]
```diff
@@ -34,7 +34,7 @@ export type XYChartProps<
   /** Total height of the desired chart svg, including margin. */
   height?: number;
   /** Margin to apply around the outside. */
-  margin?: Margin;
+  margin?: Margin | null;
```
Comment (gazcn007) [inline packages/visx-xychart/src/components/XYChart.tsx:37]: I don't think you need the null here.
You can consider converting it to `Partial<Margin>` to make it consistent?

**Pull request closed without merging**